toolchain/gcc-14.2.0.diff

```diff
@@ -0,0 +1,190 @@
+diff -ruN ../gcc-14.2.0/config.sub gcc-14.2.0/config.sub
+--- ../gcc-14.2.0/config.sub	2024-08-01 09:17:13.000000000 +0100
++++ gcc-14.2.0/config.sub	2024-12-03 03:05:07.420714106 +0000
+@@ -883,6 +883,10 @@
+ 		cpu=power
+ 		vendor=ibm
+ 		;;
++	xenon)
++		cpu=powerpc64
++		basic_os=linux
++		;;
+ 	ps2)
+ 		cpu=i386
+ 		vendor=ibm
+diff -ruN ../gcc-14.2.0/gcc/config/rs6000/altivec.md gcc-14.2.0/gcc/config/rs6000/altivec.md
+--- ../gcc-14.2.0/gcc/config/rs6000/altivec.md	2024-08-01 09:17:14.000000000 +0100
++++ gcc-14.2.0/gcc/config/rs6000/altivec.md	2024-12-03 03:14:37.514807472 +0000
+@@ -1009,7 +1009,7 @@
+ 		      (match_operand:VIshort 2 "register_operand" "v")
+                       (match_operand:V4SI 3 "register_operand" "v")]
+ 		     UNSPEC_VMSUMU))]
+-  "TARGET_ALTIVEC"
++  "(TARGET_ALTIVEC && 0)"
```

Comment:
Why disable all of these? && 0 is equivalent to false?

Reply:
Yes, this is disabling the Altivec instructions which do not exist on the Xenon CPU. All of the GCC patches do this.